julearn/pipeline/tests/test_pipeline_creator.py

```diff
@@ -155,15 +167,228 @@ def test_hyperparameter_tuning(
+    bayes_search_params : dict of str and list
+        The parameters for the search.
+
+    """
+    try:
+        from skopt import BayesSearchCV
+    except ImportError:
+        pytest.skip("skopt not installed")
+
+    pipeline, param_grid = _hyperparam_tuning_base_test(
+        X_types_iris,
+        model,
+        preprocess,
+        problem_type,
+        get_tuning_params,
+        bayes_search_params,
+    )
+    assert isinstance(pipeline, BayesSearchCV)
+    assert pipeline.search_spaces == param_grid
+
+
+def _compare_param_grids(a: Dict, b: Dict) -> None:
+    """Compare two param grids.
+
+    Parameters
+    ----------
+    a : dict
+        The first param grid (processed).
+    b : dict
+        The second param grid (raw).
+
+    Raises
+    ------
+    AssertionError
+        If the param grids are not equal.
+
+    """
+    for key, val in a.items():
+        assert key in b
+        if hasattr(val, "rvs"):
+            assert val.args[0] == b[key][0]
+            assert val.args[1] == b[key][1]
+            if b[key][2] in ["log-uniform", "loguniform"]:
+                assert val.dist.name == "loguniform"
+            elif b[key][2] == "uniform":
+                assert val.dist.name == "uniform"
+            else:
+                raise ValueError(f"Unknown distribution {b[key][2]}")
+        else:
+            assert val == b[key]
+
+
+def test_hyperparameter_tuning_distributions(
+    X_types_iris: Dict[str, List[str]],  # noqa: N803
+    model: str,
+    preprocess: Union[str, List[str]],
+    problem_type: str,
+    get_tuning_distributions: Callable,
+    search_params: Dict[str, List],
+) -> None:
+    """Test hyperparameter tuning using distributions.
+
+    Parameters
+    ----------
+    X_types_iris : dict
+        The iris dataset features types.
+    model : str
+        The model to test.
+    preprocess : str or list of str
+        The preprocessing steps to test.
+    problem_type : str
+        The problem type to test.
+    get_tuning_distributions : Callable
+        A function that returns the tuning hyperparameters for a given step.
+    search_params : dict of str and list
+        The parameters for the search.
+
+    """
+    kind = "grid"
+    if search_params is not None:
+        kind = search_params.get("kind", "grid")
+    if kind == "grid":
+        return  # No sense to test distributions for grid search
+
+    pipeline, param_grid = _hyperparam_tuning_base_test(
+        X_types_iris,
+        model,
+        preprocess,
+        problem_type,
+        get_tuning_distributions,
+        search_params,
+    )
+
+    if kind == "random":
         assert isinstance(pipeline, RandomizedSearchCV)
-        assert pipeline.param_distributions == param_grid
+        _compare_param_grids(pipeline.param_distributions,  # type: ignore
+                             param_grid)
+    else:
+        raise ValueError(f"Unknown kind {kind}")
```

Comment:
Any specific reason for raising an exception in a test function? Is it requiring something that `pytest.raises()` can provide?